### ISSUE TITLE
use standard targets for unix build process

### ISSUE THREAD

**Issue by romw** [BOINC/boinc#1328]
**Reported by ChristianB on 18 Jan 44414593 10:40 UTC**
Following the GNU convention for [makefile targets](http://www.gnu.org/prep/standards/html_node/Standard-Targets.html#Standard-Targets) I ran some tests with a fresh copy of the boinc-v2 source. I identified
several files that can be ignored and get deleted by clean/distclean and files that should be deleted by clean/distclean but are not.

Here is a short summary of the convention:

Files created by ./_autosetup need to be deleted in case of a system
upgrade because most likely the tools tested are updated. Those files
are not deleted via Makefile because it is a step before Makefile's exist.
Maybe adding an option to _autosetup that clears the files before
running the tests could help in this case.

What's important for building is that after running "make clean" we
should have the same files as after running "./configure" and after
running "make distclean" we should have the same files as after
"./_autosetup".

The attached textfile contains all those files in gitignore style and
surely some of them should go into .gitignore.

Could someone who wrote the _autosetup and/or build stuff enhance clean
and distclean please?

Migrated-From: http://boinc.berkeley.edu/trac/ticket/1357

**Comment by ChristianBeer** [comment]
Partly implemented in 977b1414727acc142a02ec4a89e172c2ef831041

remaining files from the textfile in the original ticket:
```
## should be deleted by _autosetup itself
*.in
aclocal.m4
autom4te.cache/
compile
config.guess
config.sub
configure
depcomp
install-sh
ltmain.sh
m4/libtool.m4
m4/ltoptions.m4
m4/ltsugar.m4
m4/ltversion.m4
m4/lt~obsolete.m4
missing
test-driver

### files created by configure but NOT cleaned by 'make distclean':
clientgui/res/Makefile
packages/generic/sea/Makefile
packages/solaris/CSW/Makefile
packages/solaris/CSW/boincclient/Makefile
packages/solaris/CSW/boincclient/pkginfo
packages/solaris/CSW/boincclient/prototype
packages/solaris/CSW/boincdevel/Makefile
packages/solaris/CSW/boincdevel/pkginfo
packages/solaris/CSW/boincdevel/prototype
packages/solaris/CSW/boinclibs/Makefile
packages/solaris/CSW/boinclibs/pkginfo
packages/solaris/CSW/boinclibs/prototype
packages/solaris/CSW/boincmanager/Makefile
packages/solaris/CSW/boincmanager/pkginfo
packages/solaris/CSW/boincmanager/prototype

### files created by make but NOT cleaned by 'make clean':
api/libboinc_api.a
api/libboinc_graphics2.a
api/libboinc_opencl.a
client/boinc
py/lib.linux-x86_64-2.7/
sched/libsched.a
svn_version.h
zip/libboinc_zip.a
```

**Comment by truboxl** [comment]
I feel like this will end up reimplement what `git clean -fd` do. Add `-x` to ignore `.gitignore`.
I also don't think `./_autosetup` should do the cleaning.
`make clean` or `make distclean` should handle it.
